### ISSUE TITLE
Lint `clippy::arbitrary-source-item-ordering` is too broad

### ISSUE THREAD

**Issue by eirnym** [rust-lang/rust-clippy#15573]
### Summary

This lint checks too many things at once and should be split in multiple lints. Additionally, for alphabetical order parts inside module or structure, I suggest to name it appropriately emphisising `alphabetical order` in name.

### Reproducer

I tried this code:

```rust
<code>
```

I expected to see this happen:

Instead, this happened:

### Version

```text
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: aarch64-apple-darwin
release: 1.88.0
LLVM version: 20.1.5
```
